hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/hudi/HoodieFileIndex.scala

```diff
@@ -227,7 +227,17 @@ case class HoodieFileIndex(
    */
   private def loadPartitionPathFiles(): Map[PartitionRowPath, Array[FileStatus]] = {
     val sparkEngine = new HoodieSparkEngineContext(new JavaSparkContext(spark.sparkContext))
+    val serializableConf = new SerializableConfiguration(spark.sessionState.newHadoopConf())
+
     val properties = new Properties()
+    // To support metadata listing via Spark SQL we allow users to pass the config via Hadoop Conf. Spark SQL does not
```

Comment:
Should we get these `configurations` from the `spark.sessionState.conf` for spark?

Reply:
Makes sense. It will be a better experience than having to pass it via hadoop conf. Customers would be able to enable in the spark sql session using `SET` commands.